src/App.tsx

```diff
@@ -1,24 +1,304 @@
+
+        setAllTodos(updatedTodos);
+      } catch {
+        showError(ErrorMessageType.UPDATE);
+      } finally {
+        setLoadingTodoIds([]);
+      }
+    }, [allTodos],
+  );
+
+  const changeTitleByDoubleClick = useCallback(
+    async (todoId: number, newTitle: string) => {
+      if (!newTitle.length) {
+        await handleRemoveTodo(todoId);
+
+        return;
+      }
+
+      try {
+        await setLoadingTodoIds(prev => [...prev, todoId]);
+
+        await updateTodoTitle(todoId, newTitle);
+        const newTodos = allTodos.map((todo) => {
+          if (todoId === todo.id) {
+            const updatedTodo = { ...todo };
+
+            updatedTodo.title = newTitle;
+
+            return updatedTodo;
+          }
+
+          return todo;
+        });
+
+        setAllTodos(newTodos);
+      } catch {
+        showError(ErrorMessageType.UPDATE);
+      } finally {
+        setLoadingTodoIds([]);
+      }
+    }, [allTodos],
+  );
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          <button
+            aria-label="clear"
+            type="button"
+            onClick={handleChangeStatusAllTodos}
+            className={classNames(
+              'todoapp__toggle-all',
+              {
+                active: hasActiveTodo,
+              },
+            )}
+          />
+
+          <Form
+            handleAddTodo={handleAddNewTodo}
+            todoTitle={todoTitle}
+            setTodoTitle={setTodoTitle}
+            activeInput={activeInput}
+          />
+        </header>
+
+        {(!allTodos.length && isRequest && !errorMessage)
+        && (
+          <span style={{
+            display: 'flex',
+            justifyContent: 'center',
+            color: 'grey',
+          }}
+          >
+            You have not any todos
+          </span>
+        )}
+
+        {!allTodos.length && !errorMessage && !isRequest
+          ? (
+            <Loader />
+          ) : (
+            <section className="todoapp__main">
+              <TodoList
+                todos={visibleTodos}
+                tempTodos={tempTodo}
+                handleRemoveTodo={handleRemoveTodo}
+                loadingTodoIds={loadingTodoIds}
+                handleUpdateTodoCompleted={handleUpdateTodoCompleted}
+                changeTitleByDoubleClick={changeTitleByDoubleClick}
+              />
+            </section>
+          )}
+
+        {!allTodos.length || (
```

Comment:
why don't just use `&&`?
```suggestion
        {allTodos.length && (
```